tests/test-sorting-network.cpp

```diff
@@ -0,0 +1,103 @@
+/*********************                                                        */
+/*! \file test-disjointset.cpp
+** \verbatim
+** Top contributors (to current version):
+**   Makai Mann
+** This file is part of the smt-switch project.
+** Copyright (c) 2020 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Tests for SortingNetwork.
+**
+**
+**/
+
+#include <gtest/gtest.h>
+
+#include <utility>
+#include <vector>
+
+#include "available_solvers.h"
+#include "smt.h"
+#include "sorting_network.h"
+#include "utils.h"
+
+using namespace smt;
+using namespace std;
+
+namespace smt_tests {
+
+class SortingNetworkTests
+    : public ::testing::Test,
+      public ::testing::WithParamInterface<SolverConfiguration>
+{
+ protected:
+  void SetUp() override
+  {
+    solver = create_solver(GetParam());
+    solver->set_opt("produce-models", "true");
+    solver->set_opt("incremental", "true");
+    boolsort = solver->make_sort(BOOL);
+    for (size_t i = 0; i < NUM_VARS; ++i)
+    {
+      boolvec.push_back(solver->make_symbol("b" + std::to_string(i), boolsort));
+    }
+  }
+  SmtSolver solver;
+  Sort boolsort;
+  TermVec boolvec;
+  size_t NUM_VARS = 8;
```

Comment:
I'd suggest testing also an odd length , to test well the division by 2 of the pivot. Maybe even something like `3` -- because when u divide it by 2 u get `1` and then `0`, which seem like edge cases.

Reply:
Good point, added two others to cover all cases: d6bc10dfc4f2f1fba66463bb0187347524396e1f